plutus-cbor/src/Plutus/Codec/CBOR/Encoding.hs

```diff
@@ -86,7 +87,7 @@ import PlutusTx.Builtins (subtractInteger)
 newtype Encoding = Encoding (BuiltinByteString -> BuiltinByteString)
 
 instance Semigroup Encoding where
-  (Encoding a) <> (Encoding b) = Encoding (a . b)
+  (Encoding f) <> (Encoding g) = Encoding (\(!x) -> f $ g x)
```

Comment:
We probably don't want that one... Being too strict is also not good as it prevents some compiler optimizations. Here, I'd heavily favor the composition over strictness.. I don't think it buys us much.